numpy/core/setup_common.py

```diff
@@ -104,6 +121,24 @@ def check_api_version(apiversion, codegen_dir):
         "strtoll", "strtoull", "cbrt", "strtold_l", "fallocate",
         "backtrace", "madvise"]
 
+OPTIONAL_STDFUNCS_SIGS = {}
+OPTIONAL_STDFUNCS_CALLS = {}
+
+for func in OPTIONAL_STDFUNCS:
+    set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, func, "double", ["double"])
+
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtoull", "unsigned long long", ["const char*", "char**", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtoll", "long long", ["const char*", "char**", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "strtold_l", "long double", ["const char*", "char**", "int /* locale_t */"])
+
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "pow", "double", ["double", "double"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "atan2", "double", ["double", "double"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "nextafter", "double", ["double", "double"])
+
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "madvise", "int", ["void *", "size_t", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "backtrace", "int", ["void **", "int"])
+set_sig(OPTIONAL_STDFUNCS_SIGS, OPTIONAL_STDFUNCS_CALLS, "fallocate", "int", ["int", "int", "int /* off_t */", "int /* off_t */"])
```

Comment:
Similarly, I wasn't clear if `#include "fcntl.h"` is allowed. Surely if though `fcntl.h` is missing then so is `fallocate`?